File: 56-merge-intervals/merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()
        ans = []
        st,ed = intervals[0]
        for i in range(1,len(intervals)):
            if intervals[i][0] <= ed:
                ed = max(intervals[i][1],ed)
            else :
                ans.append([st,ed])
                st,ed = intervals[i]
        ans.append([st,ed])
        st,ed = intervals[-1]
        return ans

File: 56-merge-intervals/merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(),intervals.end());
        vector<vector<int>>ans;
        vector<int>st = intervals[0];
        for(int i= 1 ; i < intervals.size();i++){
            if(st[1] >= intervals[i][0])
                st[1] = max(st[1],intervals[i][1]);
            else{
                ans.push_back(st);
                st = intervals[i];
            }
        }
        ans.push_back(st);
        return ans;
    }
};